gn3/responses/pcorrs_responses.py

```diff
@@ -0,0 +1,24 @@
+"""Functions and classes that deal with responses and conversion to JSON."""
+import json
+
+from flask import make_response
+
+class OutputEncoder(json.JSONEncoder):
+    """
+    Class to encode output into JSON, for objects which the default
+    json.JSONEncoder class does not have default encoding for.
+    """
+    def default(self, o):
+        if isinstance(o, bytes):
+            return str(o, encoding="utf-8")
+        return json.JSONEncoder.default(self, o)
+
+def build_response(data):
+    """Build the responses for the API"""
+    status_codes = {
+        "error": 400, "not-found": 404, "success": 200, "exception": 500}
+    response = make_response(
+            json.dumps(data, cls=OutputEncoder),
+            status_codes[data["status"]])
+    response.headers["Content-Type"] = "application/json"
+    return response
```

Comment:
**issue:** Doesn't flask have the method "jsonify". Why not use that? I'm not entirely sure what "data" looks like though.

Reply:
Well, yes - until you have something jsonify does not know to convert, as is the case with some of the data here, then all hell breaks loose - I tried to contain it with jsonify, and the aggravation was not worth it, so I just used the builtin json module instead